### PR TITLE
JavaScript Roundup Part 2

### PR DESCRIPTION
**Pull request by waderyan**
@gregvanl can I get an edit pass on this? Appreciate your insights and feedback. 

FYI @seanmcbreen - I tried to make this post more about extension's authors..

### CODE REVIEW

File: blogs/2016/10/31/js_roundup_2.md [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+
+1. [Decide what to build](http://code.visualstudio.com/docs/extensions/overview) - Read the various [ways to contribute](http://code.visualstudio.com/docs/extensionAPI/extension-points). Want to add your custom snippets or keybindings? You can do that easily. 
+2. [Install the extension generator](http://code.visualstudio.com/docs/tools/yocode) - Installed via yo, the starter kit starts you off on the right foot. 
+3. [Code, debug](http://code.visualstudio.com/docs/extensions/debugging-extensions), and [publish](http://code.visualstudio.com/docs/tools/vscecli) - Write code for your extension, debug using VS Code, and then publish with a single CLI command. 
+
+```zsh
+# Install the necessary tools
+npm install -g yo generator-code vsce
+
+# Install the extension generator
```
Comment (gregvanl) [inline blogs/2016/10/31/js_roundup_2.md]: Run the extension generator

File: blogs/2016/10/31/js_roundup_2.md [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+Author: Wade Anderson
+---
+
+# JavaScript Extensions Part 2
+
+October 31, 2016 by Wade Anderson, [@waderyan_](https://twitter.com/waderyan_)
+
+Visual Studio Code has excellent support for JavaScript out of the box. As with other languages, JavaScript is powered by a language service. The [JavaScript language service](https://github.com/Microsoft/TypeScript/wiki/Salsa) is implemented by the TypeScript team, allowing JavaScript developers to leverage the best IntelliSense experience. 
+
+But what other features can you get with VS Code? Because of VS Code's rich extensibilty model, there are many features extensions can provide. This post is a follow up to my previous post about JavaScript extensions and a brief introduction to writing your own JavaScript extension.
```
Comment (gregvanl) [inline blogs/2016/10/31/js_roundup_2.md]: VS Code has a rich extensibility model and many features are provided through VS Code extensions.

File: blogs/2016/10/31/js_roundup_2.md [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+
+Internally, VS Code uses [js-beautify](https://www.npmjs.com/package/js-beautify). This extension allows you to specify a `.jsbeautifyrc` file to indicate the formatting style for your JavaScript, CSS, Sass, and HTML code. 
+
+You can search the Marketplace for more formatters using the new [Formatters category](https://marketplace.visualstudio.com/search?target=VSCode&category=Formatters&sortBy=Downloads). 
+
+>**Note:** For extension authors, we are working on a blog post for Formatting best practices. Stay tuned as it will be coming soon. 
+
+# Keymaps for Sublime Text and Atom
+
+There are two extensions in this section, one for [Atom](https://marketplace.visualstudio.com/items?itemName=ms-vscode.atom-keybindings) and another for [Sublime Text](https://marketplace.visualstudio.com/items?itemName=ms-vscode.sublime-keybindings). If you have used these editors you have likely developed habits with your keyboards shortcuts. These extensions bring the keyboard shortcuts from Atom and Sublime Text to VS Code.
```
Comment (gregvanl) [inline blogs/2016/10/31/js_roundup_2.md]: Can we just say "Sublime" on the later uses of "Sublime Text". Sublime Text is cumbersome and most people just say "Sublime"

File: blogs/2016/10/31/js_roundup_2.md [hunk 4]
```diff
@@ -0,0 +1,133 @@
...
+    "mac": "<keyboard shortcut for mac>",
+    "linux": "<keyboard shortcut for linux",
+    "win": "<keyboard shortcut for windows",
+    "key": "<default keyboard shortcut>",
+    "command": "<name of the command in VS Code"
+}
+```
+
+If you find that VS Code does not have a feature from Atom or Sublime Text, please comment in these Github issues ([Atom](https://github.com/waderyan/vscode/issues/14316) and [Sublime Text](https://github.com/Microsoft/vscode/issues/3776)). Additionally, there are many extensions in the Marketplace to add the missing features. 
+
```
Comment (gregvanl) [inline blogs/2016/10/31/js_roundup_2.md]: Give this another H2 header. Something like "Extension grab bag" or "It's raining extensions".

File: blogs/2016/10/31/js_roundup_2.md [hunk 5]
```diff
@@ -0,0 +1,133 @@
...
+* [Join Lines](https://marketplace.visualstudio.com/items?itemName=wmaurer.join-lines)
+* [Paste and Indent](https://marketplace.visualstudio.com/items?itemName=Rubymaniac.vscode-paste-and-indent)
+* [FontSize Shortcuts](https://marketplace.visualstudio.com/items?itemName=peterjuras.fontsize-shortcuts)
+* [Bracket Selection](https://marketplace.visualstudio.com/items?itemName=guosong.bracketselection)
+* [change case](https://marketplace.visualstudio.com/items?itemName=wmaurer.change-case)
+* [expand-region](https://marketplace.visualstudio.com/items?itemName=letrieu.expand-region)
+* [transpose](https://marketplace.visualstudio.com/items?itemName=v4run.transpose)
+* [Close HTML/XML tag](https://marketplace.visualstudio.com/items?itemName=Compulim.compulim-vscode-closetag)
+
+Do you have other editors or IDEs you'd like to make a keymap for? Simply follow the instructions in the [`contribrutions.keybindings` section](http://code.visualstudio.com/docs/extensionAPI/extension-points#_contributeskeybindings) and the [Customization -> Keybindings document](http://code.visualstudio.com/docs/customization/keybindings).
```
Comment (gregvanl) [inline blogs/2016/10/31/js_roundup_2.md]: Move this up to be directly after Atom and Sublime section.

File: blogs/2016/10/31/js_roundup_2.md (hunk 1 above)
Comment (waderyan) [inline blogs/2016/10/31/js_roundup_2.md]: Resolved

File: blogs/2016/10/31/js_roundup_2.md [hunk 6]
```diff
@@ -0,0 +1,133 @@
...
+
+But what other features can you get with VS Code? Because of VS Code's rich extensibilty model, there are many features extensions can provide. This post is a follow up to my previous post about JavaScript extensions and a brief introduction to writing your own JavaScript extension. 
+
+## Do you want to build an extension? 
+
+You can build an extension with JavaScript. To write your first extension takes only three simpe steps. 
+
+1. [Decide what to build](http://code.visualstudio.com/docs/extensions/overview) - Read the various [ways to contribute](http://code.visualstudio.com/docs/extensionAPI/extension-points). Want to add your custom snippets or keybindings? You can do that easily. 
+2. [Install the extension generator](http://code.visualstudio.com/docs/tools/yocode) - Installed via yo, the starter kit starts you off on the right foot.
```
Comment (waderyan) [inline blogs/2016/10/31/js_roundup_2.md]: Resolved

File: blogs/2016/10/31/js_roundup_2.md (hunk 2 above)
Comment (waderyan) [inline blogs/2016/10/31/js_roundup_2.md]: Done.

File: blogs/2016/10/31/js_roundup_2.md (hunk 3 above)
Comment (waderyan) [inline blogs/2016/10/31/js_roundup_2.md]: I had a discussion with Daniel about this. We should use Sublime Text.

File: blogs/2016/10/31/js_roundup_2.md (hunk 4 above)
Comment (waderyan) [inline blogs/2016/10/31/js_roundup_2.md]: Great suggestion!

File: blogs/2016/10/31/js_roundup_2.md (hunk 5 above)
Comment (waderyan) [inline blogs/2016/10/31/js_roundup_2.md]: Done.

File: blogs/2016/10/31/js_roundup_2.md [hunk 7]
```diff
@@ -0,0 +1,133 @@
...
+
+```zsh
+# Install the necessary tools
+npm install -g yo generator-code vsce
+
+# Install the extension generator
+yo code
+
+# Write code, debug, then publish
+vsce publish
```
Comment (waderyan) [inline blogs/2016/10/31/js_roundup_2.md]: resolved

File: blogs/2016/10/31/js_roundup_2.md [hunk 8]
```diff
@@ -0,0 +1,133 @@
...
+```
+
+You can use the extensions below as a model for how to build your own and install them to accelerate your JavaScript coding productivity.
```
Comment (waderyan) [inline blogs/2016/10/31/js_roundup_2.md]: resolved

**Comment by gregvanl** [comment]
The updates look good. When do you want this to go out, Monday? I'll hold on merging into master since I'd like to publish the keyboard shortcut PDFs over the weekend.

**Comment by waderyan** [comment]
Yes let's get this out today. We can discuss in our iteration meeting.

**Pull request merged**